multiraft/multiraft.go

```diff
@@ -758,6 +785,20 @@ func (s *state) propose(p *proposal) {
 		s.removePending(nil /* group */, p, ErrGroupDeleted)
 		return
 	}
+
+	found := false
+	for _, nodeID := range g.nodeIDs {
+		if nodeID == s.nodeID {
+			found = true
+			break
+		}
+	}
+	if !found {
+		// If we are not a member of the group, don't allow any proposals.
+		s.removePending(nil, p, ErrGroupDeleted)
```

Comment:
I don't think so; if logging is appropriate it will happen at the level that handles the ErrGroupDeleted that we're returning here.